packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2507,6 +2507,21 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
 
   Element _parent;
 
+  // Custom implementation of `operator ==` optimized for the ".of" pattern
+  // used with `InheritedWidgets`.
+  @override
+  bool operator ==(Object other) => identical(this, other);
+
+  // Custom implementation of hash code optimized for the ".of" pattern
+  // used with `InheritedWidgets`.
+  //
+  // `Element.inheritFromWidgetOfExactType` relies heavily on hash-based
+  // `Set` look-ups, putting this getter on the performance critical path.
+  @override
+  int get hashCode => _cachedHash;
+  final int _cachedHash = _nextHashCode = (_nextHashCode + 1) % 0xffffff;
```

Comment:
why only 24 bits?

Reply:
Copied from https://github.com/dart-lang/sdk/issues/31011#issuecomment-344245320, but I'm guessing this is so it fits within the [SMI representation](https://www.dartlang.org/articles/dart-vm/numeric-computation).

Reply:
Adding a comment with explanation.

Reply:
We're about to switch to 64bit ints, which might make this moot.
cc @floitschG for advice.